image-loader/app/lib/Downloader.scala

```diff
@@ -25,14 +25,14 @@ class Downloader(implicit ec: ExecutionContext) {
     val request = new Request.Builder().url(uri.toString).build()
     val response = client.newCall(request).execute()
 
-    val maybeExpectedSize = Try{response.header("Content-Length").toInt}.toOption
+    val maybeExpectedSize = Try{response.header("Content-Length").toInt}
 
     maybeExpectedSize match {
-      case None => {
-        Logger.error(s"Missing content-length header from $uri")
+      case Failure(exception) => {
+        Logger.error(s"Missing content-length header from $uri", exception)
```

Comment:
Do you need something like `.message` or `.getMessage` on the exception?

Reply:
The type signature suggests this is possible. I imagine this allows the log line to include the stack trace.